### ISSUE TITLE
Incorrect platform detection in process.py

### ISSUE THREAD

**Issue by kevinrtai** [Lux-AI-Challenge/Lux-Design-S2#215]
https://github.com/Lux-AI-Challenge/Lux-Design-S2/blob/87cd543343f1abd80367b68af797a2c67c46c3c7/luxai_s2/luxai_runner/process.py#L57

This doesn't work on Mac OS b/c `sys.platform` is darwin on Mac.

**Comment by StoneT2000** [comment]
That should be fine since mac is unix

**Comment by kevinrtai** [comment]
Hmm, I get the following error when I try to run a non-python on MacOS but I don't have the same problem on my Ubuntu machine.

```
kevinrtai@kevinrtai-mbp:~/projects/luxai-s2/bots
$ luxai-s2 /Users/kevinrtai/projects/luxai-s2/bots/run-agent.out ../Lux-Design-S2/kits/python/main.py --output replay.json
Task exception was never retrieved
future: <Task finished name='Task-3' coro=<BotProcess.start() done, defined at /Users/kevinrtai/.pyenv/versions/3.10.7/envs/luxai-s2/lib/python3.10/site-packages/luxai_runner/process.py:45> exception=FileNotFoundError(2, 'No such file or directory')>
Traceback (most recent call last):
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/envs/luxai-s2/lib/python3.10/site-packages/luxai_runner/process.py", line 56, in start
    self._agent_process = await asyncio.create_subprocess_exec(
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/subprocess.py", line 218, in create_subprocess_exec
    transport, protocol = await loop.subprocess_exec(
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/base_events.py", line 1667, in subprocess_exec
    transport = await self._make_subprocess_transport(
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/unix_events.py", line 207, in _make_subprocess_transport
    transp = _UnixSubprocessTransport(self, protocol, args, shell,
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/base_subprocess.py", line 36, in __init__
    self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/unix_events.py", line 799, in _start
    self._proc = subprocess.Popen(
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/subprocess.py", line 969, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/subprocess.py", line 1845, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/kevinrtai/projects/luxai-s2/bots\\run-agent.out'
Traceback (most recent call last):
  File "/Users/kevinrtai/.pyenv/versions/luxai-s2/bin/luxai-s2", line 8, in <module>
    sys.exit(main())
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/envs/luxai-s2/lib/python3.10/site-packages/luxai_runner/cli.py", line 161, in main
    asyncio.run(eps.run())
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/envs/luxai-s2/lib/python3.10/site-packages/luxai_runner/episode.py", line 144, in run
    resolved_actions = await asyncio.gather(*action_coros)
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/envs/luxai-s2/lib/python3.10/site-packages/luxai_runner/bot.py", line 79, in step
    action, stderr = await asyncio.wait_for(
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
    return fut.result()
  File "/Users/kevinrtai/.pyenv/versions/3.10.7/envs/luxai-s2/lib/python3.10/site-packages/luxai_runner/process.py", line 106, in write
    self._agent_process.stdin.write(msg.encode())
AttributeError: 'BotProcess' object has no attribute '_agent_process'
```

I think it's exactly because MacOS is a Unix platform that this doesn't work. It's trying to use a backslash as a file path separator (like on Windows) when it should be using forward slash.

`FileNotFoundError: [Errno 2] No such file or directory: '/Users/kevinrtai/projects/luxai-s2/bots\\run-agent.out'`

**Comment by rooklift** [comment]
To be clear, the issue is the string "darwin" contains "win"

**Comment by kevinrtai** [comment]
Yes

**Comment by StoneT2000** [comment]
ah I see! Will fix

**Comment by StoneT2000** [comment]
fixed in v2.1.8
